Common/itkTransformixInputPointFileReader.hxx

```diff
@@ -40,30 +40,60 @@ TransformixInputPointFileReader<TOutputMesh>::GenerateOutputInformation()
   {
     this->m_Reader.close();
   }
-  this->m_Reader.open(this->m_FileName);
 
-  /** Read the first entry */
-  std::string indexOrPoint;
-  this->m_Reader >> indexOrPoint;
+  if( this->m_Binary ) {
+    this->m_Reader.open(this->m_FileName, std::ios::binary);
+
+    /** Read index flag */
+    unsigned long isindex;
```

Comment:
I'm sorry to say the use of (signed or unsigned) `long` is problematic, because its size differs between currently supported platforms and compilers. On Linux/GCC it's typically 64 bits, whereas on Windows/MSVC, it's 32 bits.  If you want to write unsigned integers on one platform and read them back on another, better use either `uint32_t` or `uint64_t`. (In general use cases, `unsigned int` is usually fine, as it is the "default" unsigned type, and it's 32-bits on all supported platforms.)